rust/signed_doc/src/signature/kid/role.rs

```diff
@@ -0,0 +1,42 @@
+//! COSE Signature Protected Header `kid` URI Catalyst User Role.
+
+use std::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+/// Project Catalyst User Role associated with the signature.
+///
+/// <https://github.com/input-output-hk/catalyst-CIPs/blob/x509-catalyst-role-definitions/CIP-XXXX/README.md>
+#[repr(u16)]
+#[derive(Debug, Copy, Clone)]
+pub enum Role {
```

Comment:
Don't restrict this to an enum, it should be general over any positive integer.
Suggest instead just:
```
pub struct Role(u16)
```
16 bits should be more than enough role numbers.